DataFormats/Portable/interface/PortableDeviceCollection.h

```diff
@@ -8,37 +8,115 @@
 #include "HeterogeneousCore/AlpakaInterface/interface/config.h"
 #include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
 #include "HeterogeneousCore/AlpakaInterface/interface/traits.h"
+#include "DataFormats/Portable/interface/PortableCollectionCommon.h"
 
 // generic SoA-based product in device memory
-template <typename T, typename TDev, typename = std::enable_if_t<cms::alpakatools::is_device_v<TDev>>>
+template <typename TDev,
+          typename T0,
+          typename T1 = void,
+          typename T2 = void,
+          typename T3 = void,
+          typename T4 = void,
```

Comment:
The `PortableDeviceCollection` is still listed in `DataFormats/PortableTestObjects/src/alpaka/classes_cuda_def.xml` and hence added to the dictionary (is it not needed?). This is the breaking step.

And as soon as the class is a product, we need a dictionary (otherwise we fail to register the product at runtime).

Reply:
Thanks, so the problem is already in the part of dictionaries (that framework uses for reflection), and not just the serialization.